mlflow/transformers.py

```diff
@@ -0,0 +1,851 @@
+             ``transformers`` flavor. Calls to :py:func:`save_model()` and :py:func:`log_model()`
+             produce a pip environment that contain these requirements at a minimum.
+    """
+    from transformers import TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel
+
+    if not isinstance(model, (TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel)):
+        raise MlflowException(
+            "The supplied model type is unsupported. The model must be one of: PreTrainedModel, "
+            "TFPreTrainedModel, or FlaxPreTrainedModel",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    base_reqs = ["transformers"]
+    base_reqs.extend(_model_packages(model))
+    return [_get_pinned_requirement(module) for module in base_reqs]
+
+
+def _validate_transformers_model_dict(transformers_model):
+    """
+    Validator for a submitted save dictionary for the transformers model. If any additional keys
+    are provided, raise to indicate which invalid keys were submitted.
+    """
+    if isinstance(transformers_model, dict):
+        invalid_keys = [key for key in transformers_model.keys() if key not in _SUPPORTED_SAVE_KEYS]
+        if invalid_keys:
+            raise MlflowException(
+                "Invalid dictionary submitted for 'transformers_model'. The "
+                f"key(s) {invalid_keys} are not permitted. Must be one of: "
+                f"{_SUPPORTED_SAVE_KEYS}",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        if _MODEL_KEY not in transformers_model:
+            raise MlflowException(
+                f"The 'transformers_model' dictionary must have an entry for {_MODEL_KEY}",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+
+@experimental
+def get_default_conda_env(model):
+    """
+    :return: The default Conda environment for MLflow Models produced with the ``transformers``
+             flavor, based on the model instance framework type of the model to be logged.
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements(model))
+
+
+@experimental
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def save_model(
+    transformers_model,
+    path: str,
+    processor=None,
+    task: Optional[str] = None,
+    model_card=None,
+    code_paths: Optional[List[str]] = None,
+    mlflow_model: Optional[Model] = None,
+    signature: Optional[ModelSignature] = None,
+    input_example: Optional[ModelInputExample] = None,
+    pip_requirements: Optional[Union[List[str], str]] = None,
+    extra_pip_requirements: Optional[Union[List[str], str]] = None,
+    conda_env=None,
+    metadata: Dict[str, Any] = None,
+    **kwargs,
+) -> None:
+    """
+    Save a trained transformers model to a path on the local file system.
+
+    :param transformers_model: A trained transformers `Pipeline` or a dictionary that maps
+                               required components of a pipeline to the named keys of
```

Comment:
We can insert a newline if we need more space:

```suggestion
    :param transformers_model:
        A trained transformers `Pipeline` or a dictionary that maps
        required components of a pipeline to the named keys of
```

Reply:
Good idea! Thanks :)